### ISSUE TITLE
Inconsistent error reporting in .vue files containing typescript

### ISSUE THREAD

**Issue by ma7moudat** [vuejs/vue-eslint-parser#57]
**I need to say in advance that I found an alternative (see below), but I think the problem might really throw someone off and cost them time, so better take a look at it. 
And I'm not sure it really belongs here tbh! So tell me to report it somewhere else if needed.**

### Steps to reproduce:

I created a new @vue/cli project (using v3.7.0) with typescript and eslint (AirBnB preset). Everything was working fine until I added the following code to one of my components:

```vue
<template>
    <div class="o-manifest">
        
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { CartHttpProvider } from '@/providers';
import { IManifest } from '@/models';
import { logger } from '@/util';

@Component
export default class OManifest extends Vue {
  private model:IManifest = <IManifest>{};

  async mounted() {
    const response = await CartHttpProvider.cartRead();
    this.model = response.data;
    logger.debug(this.model);
  }
}
</script>
```

which caused the following error

```console
 ERROR  Failed to compile with 1 errors                         6:15:01 PM

 error  in ./src/components/organisms/Manifest.vue

Module Error (from ./node_modules/eslint-loader/index.js):
error: Parsing error: Expression expected at src/components/organisms/Manifest.vue:19:4:
  17 | 
  18 |   async mounted() {
> 19 |     const response = await CartHttpProvider.cartRead();
     |    ^
  20 |     this.model = response.data;
  21 |     logger.debug(this.model);
  22 |   }


1 error found.

 @ ./src/components/organisms/index.ts 1:0-39 2:0-21
 @ ./src/components/index.ts
 @ ./src/main.ts
 @ multi ./src/main.ts

Type checking in progress...
No type errors found
Version: typescript 3.5.3
Time: 1009ms
```

If I restart the npm `serve` task it works the first time, but any change I make causes the error again.

### Workaround

It was solved by changing the typecasting
```
private model:IManifest = <IManifest>{};
```
to
```
private model:IManifest = {} as IManifest;
```

**Comment by braian125** [comment]
Why are there no comments about this? it is a very annoying problem.

I have tried to find a solution and there is no precise documentation

**Comment by beamsies** [comment]
Same issues here on my end.

**Comment by ota-meshi** [comment]
If you want to use `<>` style typecasts, you'll need to tell `@typescript-eslint/parser` not to use JSX.
Can you try this by changing the `jsx` option?

**Comment by ota-meshi** [comment]
Try changing the option.
